### ISSUE TITLE
what does lgt_reg mean, is it the entropy maximization term H(q)? 

### ISSUE THREAD

**Issue by victor000000** [Philip-Bachman/amdim-public#7]
at costs.py line54-59

          # compute src->trg raw scores for batch on this gpu
          raw_scores = torch.mm(r_src, r_trg).float()
          raw_scores = raw_scores.reshape(n_batch_gpu, n_batch, n_locs)
          raw_scores = raw_scores / n_rkhs ^ 0.5
          lgt_reg = 5e-2 * (raw_scores ^ 2.).mean()
          raw_scores = tanh_clip(raw_scores, clip_val=self.tclip)

what dose lgt_reg mean?
how to calculate entropy maximization term H(q) of Eqn.(6) in the paper?

**Comment by Philip-Bachman** [comment]
The mixture-based representation code is not currently part of this repo, so there are no entropy terms in the costs implemented here. The lgt_reg term is the regularization on the NCE logits mentioned in the last paragraph of Section 3.3 of the paper (arxiv v2, from July 8th).

The aim of the logit regularization and tanh clipping is to limit the dynamic range of the NCE logits, thereby mitigating some gradient spikes and forcing the encoder to be more efficient about how it arranges the representations in the latent space. This is similar to the annealed cosine similarities used in other recent contrastive representation learning papers, where the temperature parameter controls the dynamic range of the NCE logits.

**Comment by victor000000** [comment]
Thanks, it's clear
